recognition.py

```diff
@@ -0,0 +1,127 @@
+    result = cv2.bitwise_and(videostream, videostream, mask=fgmask)
+    return result
+
+
+def findFingers(result,drawing):  # -> finished bool, count: finger count
+    #  convexity defect
+    hull = cv2.convexHull(result, returnPoints=False)
+    if len(hull) > 3:
+        defects = cv2.convexityDefects(result, hull)
+        if type(defects) != type(None):
+            count = 0
+            for i in range(defects.shape[0]):  # calculate the angle
+                s, e, f, d = defects[i][0]
+                start = tuple(result[s][0])
+                end = tuple(result[e][0])
+                far = tuple(result[f][0])
+                a = math.sqrt((end[0] - start[0]) ** 2 + (end[1] - start[1]) ** 2)
+                b = math.sqrt((far[0] - start[0]) ** 2 + (far[1] - start[1]) ** 2)
+                c = math.sqrt((end[0] - far[0]) ** 2 + (end[1] - far[1]) ** 2)
+                angle = math.acos((b ** 2 + c ** 2 - a ** 2) / (2 * b * c))  # cosine theorem
+                if angle <= math.pi / 2:  # angle less than 90 degree, treat as fingers
+                    count += 1
+                    cv2.circle(drawing, far, 8, [211, 84, 0], -1)
+            return True, count
+    return False, 0
+
+# Do camera operations
+camera_input = int(input('Enter camera number: '))
+camera = cv2.VideoCapture(camera_input)
+camera.set(10,200)
+font = cv2.FONT_HERSHEY_SIMPLEX
+cv2.namedWindow('trackbar')
+cv2.createTrackbar('trh1', 'trackbar', threshold, 100, printThreshold)
+print("Press b to capture background & begin detection or r to reset")
+
+
+while camera.isOpened():
+    ret, videostream = camera.read()
+    threshold = cv2.getTrackbarPos('trh1', 'trackbar')
+    videostream = cv2.bilateralFilter(videostream, 5, 50, 100)  # smoothing filter
+    videostream = cv2.flip(videostream, 1)  # flip the videostream horizontally
+    cv2.rectangle(videostream, (int(capture_region_x * videostream.shape[1]), 0),
+                 (videostream.shape[1], int(capture_region_y * videostream.shape[0])), (255, 0, 0), 2)
+    cv2.imshow('original', videostream)
+
+    #  Main operation
+    if isBgCaptured == 1:  # Only runs once background is captured
+        img = removeBG(videostream)
+        img = img[0:int(capture_region_y * videostream.shape[0]),
+                    int(capture_region_x * videostream.shape[1]):videostream.shape[1]]  # clip the ROI
+        #cv2.imshow('mask', img) #show mask image
+
+        # convert the image into binary image
+        gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        blur = cv2.GaussianBlur(gray, (blurValue, blurValue), 0)
+        #cv2.imshow('blurred', blur) # show blur image
+        ret, thresh = cv2.threshold(blur, threshold, 255, cv2.THRESH_BINARY)
+        #cv2.imshow('binary', thresh) # show binary image
+
+
+        # get the coutours
+        new_threshold = copy.deepcopy(thresh)
+        img2,contours, hierarchy = cv2.findContours(new_threshold, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
```

Comment:
What does cv2.CHAIN_APPROX_SIMPLE do?

Reply:
"Do you need all the points on the line to represent that line? No, we need just two end points of that line. This is what cv2.CHAIN_APPROX_SIMPLE does." From https://docs.opencv.org/3.1.0/d4/d73/tutorial_py_contours_begin.html